mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -241,52 +241,51 @@ class Completions(
     )
   end isAbstractType
 
-  private def findSuffix(symbol: Symbol): Option[String] =
-    val bracketSuffix =
-      if shouldAddSnippet &&
-        cursorPos.allowBracketSuffix && symbol.info.typeParams.nonEmpty
-      then "[$0]"
-      else ""
-
-    val bracesSuffix =
-      if shouldAddSnippet && symbol.is(Flags.Method)
-      then
-        val paramss = getParams(symbol)
-        paramss match
-          case Nil => ""
-          case List(Nil) => "()"
-          case _ if config.isCompletionSnippetsEnabled =>
-            val onlyParameterless = paramss.forall(_.isEmpty)
-            lazy val onlyImplicitOrTypeParams = paramss.forall(
-              _.exists { sym =>
-                sym.isType || sym.is(Implicit) || sym.is(Given)
-              }
-            )
-            if onlyParameterless then "()" * paramss.length
-            else if onlyImplicitOrTypeParams then ""
-            else if bracketSuffix == "[$0]" then "()"
-            else "($0)"
-          case _ => ""
-        end match
-      else ""
-
-    val templateSuffix =
-      if shouldAddSnippet && cursorPos.allowTemplateSuffix
-        && isAbstractType(symbol)
-      then
-        if bracketSuffix.nonEmpty || bracesSuffix.contains("$0") then " {}"
-        else " {$0}"
-      else ""
-
-    val concludedSuffix = bracketSuffix + bracesSuffix + templateSuffix
-    if concludedSuffix.nonEmpty then Some(concludedSuffix) else None
+  private def findSuffix(symbol: Symbol): CompletionSuffix =
+    CompletionSuffix.empty
+      .map { suffix => // for [] suffix
+        if shouldAddSnippet &&
+          cursorPos.allowBracketSuffix && symbol.info.typeParams.nonEmpty
+        then suffix.copy(bracket = true, snippet = SuffixKind.Bracket)
+        else suffix
+      }
+      .map { suffix => // for () suffix
```

Comment:
Having map here is a bit weird, since I would expect some kind of collection to be used. Could we stay with the `if-else` staements?

Reply:
what do you say if I rename `map` to something like `chain` or something?

---

I don't want to stay `if-else` because if we write in that way it's gonna be something like

```scala
val empty = CompletionSuffix.empty
val bracketSuffix = { if .. empty.copy(...) else ... }
val braceSuffix = { if ... bracketSuffix.copy(...) else ...}
val templateSuffix = { if ... braceSuffix.copy(...) else ... }
templateSuffix
```

and it's error-prone because we can access `bracketSuffix` from `templateSuffix = {...}` where it should only access `braceSuffix`.

Reply:
ok sure, that's also fine.

Reply:
renamed https://github.com/scalameta/metals/pull/4412/commits/ac8d8e90e2597084f30b28848019cbca3618324b